api/src/data_migration/load/load_oracle_data_task.py

```diff
@@ -0,0 +1,176 @@
+#
+# Load data from legacy (Oracle) tables to staging tables.
+#
+
+import logging
+import time
+
+import sqlalchemy
+
+import src.db.foreign
+import src.db.models.staging
+import src.logging
+import src.task.task
+from src.adapters import db
+from src.util import datetime_util
+
+from . import sql
+
+logger = logging.getLogger(__name__)
+
+
+class LoadOracleDataTask(src.task.task.Task):
+    """Task to load data from legacy tables to staging tables."""
+
+    def __init__(
+        self,
+        db_session: db.Session,
+        foreign_tables: dict[str, sqlalchemy.Table],
+        staging_tables: dict[str, sqlalchemy.Table],
+    ) -> None:
+        if foreign_tables.keys() != staging_tables.keys():
+            raise ValueError("keys of foreign_tables and staging_tables must be equal")
+
+        super().__init__(db_session)
+        self.foreign_tables = foreign_tables
+        self.staging_tables = staging_tables
+
+    def run_task(self) -> None:
+        """Main task process, called by run()."""
+        with self.db_session.begin_nested():
+            self.log_database_settings()
+        self.load_data()
+
+    def log_database_settings(self) -> None:
+        """Log database settings related to foreign tables for easier troubleshooting."""
+        metadata = sqlalchemy.MetaData()
+        engine = self.db_session.bind
+
+        # Use reflection to define built-in views as Table objects.
+        foreign_servers = sqlalchemy.Table(
+            "foreign_servers", metadata, autoload_with=engine, schema="information_schema"
+        )
+        foreign_server_options = sqlalchemy.Table(
+            "foreign_server_options", metadata, autoload_with=engine, schema="information_schema"
+        )
+
+        logger.info(
+            "foreign server settings",
+            extra={
+                "foreign_servers": self.db_session.execute(
+                    sqlalchemy.select(foreign_servers)
+                ).all(),
+                "foreign_server_options": self.db_session.execute(
+                    sqlalchemy.select(foreign_server_options)
+                ).all(),
+            },
+        )
+
+    def load_data(self) -> None:
+        """Load the data for all tables defined in the mapping."""
+        for table_name in self.foreign_tables:
+            try:
+                with self.db_session.begin_nested():
+                    self.load_data_for_table(table_name)
+                    self.db_session.commit()
```

Comment:
I think if you're using begin_nested, the commit should happen automatically as part of the context manager:
https://docs.sqlalchemy.org/en/20/orm/session_transaction.html#using-savepoint

I do think we should figure out what is going on with the issue you mentioned before when not using `begin_nested`. As you currently have it written, there isn't a `commit` for the overall session so I think that implicitly happens when the program exits?